modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -87,15 +97,24 @@ def report msg, response, error_only
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
 
-    def match_zone(fqdn)
+    def match_zone(fqdn_or_ip)
       dns_zones = enum_zones
       weight = 0 # sub zones might be independent from similar named parent zones; use weight for longest suffix match
       matched_zone = nil
+      ip_check = validate_ip6(fqdn_or_ip)
```

Comment:
Please don't do this: nil is also evaluated to false, anything but nil evaluates as true, this is going to cause issues. As I suggested, remove this conditional altogether and pass in the zone name as a parameter.